### DISCUSSION TITLE
Comictagger plugin

### DISCUSSION THREAD

**Discussion by mizaki** [bpepple/metron#179] [Q&A] [answered]
I've updated the [plugin](https://github.com/mizaki/comictagger-metron-talker/tree/master) with a couple of new options and wanted to make you are aware and make sure there is no concern about hitting the API. It uses mokkari so it's following the rate limit. All these options default to off.

- Series cover support. As there is no series image, when a series is active in the series window, it gets the full issue list for the series and tries for the issue 1 cover. (1 call to /series/id and 1 call to /issue)
- Variant cover support in the issue window. (1 call to /issue)
- Variant cover matching support. This is used when auto-tagging. Using the issue number a list of possible issues are created. For each issue a call to /issue is made. It should always be limited but say if it was vague like "Batman" and issue 1, that could be quite a few results.

I think only the last one could cause concern but maybe more for the user hitting the limiting rather than upsetting the API.

Happy to remove any if you believe it will be a problem.

**Comment by bpepple** [comment, marked as answer]
> I've updated the [plugin](https://github.com/mizaki/comictagger-metron-talker/tree/master) with a couple of new options and wanted to make you are aware and make sure there is no concern about hitting the API. It uses mokkari so it's following the rate limit. All these options default to off.
> 
>     * Series cover support. As there is no series image, when a series is active in the series window, it gets the full issue list for the series and tries for the issue 1 cover. (1 call to /series/id and 1 call to /issue)
> 
>     * Variant cover support in the issue window. (1 call to /issue)

I don't see a problem with these.

>     * Variant cover matching support. This is used when auto-tagging. Using the issue number a list of possible issues are created. For each issue a call to /issue is made. It should always be limited but say if it was vague like "Batman" and issue 1, that could be quite a few results.
> 
> 
> I think only the last one could cause concern but maybe more for the user hitting the limiting rather than upsetting the API.

Yeah, this feature I'm a bit dubious about based on resource usage to benefit provided. Depending on the numbers of users it could potentially increase my server costs for something in my opinion isn't worth attempting to work around a user's janky comic. I'd prefer this feature to be removed, unless you guys (CT devs) want to cover any potential server costs increase. 😱

**Reply by mizaki** [reply]
No problem, I'll remove that option. And, of course, if any issues arise, you know where to shout :)

**Reply by bpepple** [reply]
Cool, Thanks for the work on this.

**Comment by mizaki** [comment]
I've added a blurb/about to the settings page and have taken the text from the web page. Are you happy with this text or would you prefer something else?
![image](https://github.com/bpepple/metron/assets/1141189/6f346daf-86f5-4c6e-92fe-f48b334e29af)

**Reply by bpepple** [reply]
@mizaki That looks fine to me.

**Comment by mizaki** [comment]
Here are the changes adding the donation information if you want to approve them?

https://github.com/comictagger/metron_talker/pull/49/files

README: `Please support Metron's costs and further development by [donating](https://opencollective.com/metron) if you are able, thank you.`

Settings window
![image](https://github.com/user-attachments/assets/5288df80-86ca-4ca5-95a6-079162133949)

Series and issues list
![image](https://github.com/user-attachments/assets/f5c938ee-2eb4-4fab-b1f9-01fd33d60d3e)

**Reply by bpepple** [reply]
That looks fine.  Thx!
